resources/assets/react/screens/Profile/Contributions.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Display all contributions of a given user.
+ *
+ * @author Amaury Guichard <amaury.guichard@etu.utc.fr>
+ *
+ * @copyright Copyright (c) 2019, SiMDE-UTC
+ * @license GNU GPL-3.0
+ */
+
+import React from 'react';
+import { connect } from 'react-redux';
+import moment from 'moment';
+import ContributionsCard from '../../components/Profile/ContributionsCard';
+
+import actions from '../../redux/actions';
+
+@connect(store => ({
+	config: store.config,
+	contributions: store.getData(['user', 'contributions']),
+	contributionsFetching: store.isFetching(['user', 'contributions']),
+	contributionsFetched: store.isFetched(['user', 'contributions']),
+}))
+class Contributions extends React.Component {
+	componentDidMount() {
+		const { dispatch, contributionsFetched, contributionsFetching } = this.props;
+
+		if (!contributionsFetched && !contributionsFetching) {
+			dispatch(actions.user.contributions.all());
+		}
+	}
+
+	render() {
+		const { contributions, contributionsFetched } = this.props;
+		return (
+			<div>
+				{contributionsFetched &&
+					contributions.map(contribution => {
+						return (
+							<ContributionsCard
+								key={contribution.start}
+								semester1={contribution.semesters[0].name}
+								semester2={
+									contribution.semesters.length === 2 ? contribution.semesters[1].name : ''
+								}
+								amount={contribution.amount}
+								start={moment(contribution.start, 'YYYY-MM-DD').format('Do MMMM YYYY')}
+								end={moment(contribution.end, 'YYYY-MM-DD').format('Do MMMM YYYY')}
```

Comment:
I think it would be better to pass the moment object to the component if we want to enhance and re-use it. Then the formatting could be done inside the component.

Reply:
so it would be start={moment(contribution.start, 'YYYY-MM-DD')} ? or start={contribution.start} then I would create the moment in the component ?

Reply:
I think we can use `start={moment(contribution.start, 'YYYY-MM-DD')}`

Reply:
@NastuzziSamy ?

Reply:
Done.